### PR TITLE
Scalars tutorial for TensorBoard and Keras 

### PR DESCRIPTION
**Pull request by manivaradarajan**
This is a very quick introduction to viewing default scalar summaries when developing a Keras model in TensorFlow. It's intended to be very simple to show that virtually no extra code is required to get started using TensorBoard to see how loss is changing during training.

To some extent this repeats some of the material in the "Quick Start", but since it's very focussed I don't think the repetition is a bad thing.

### CODE REVIEW

**Comment by GalOshri** [comment]
Thank you for adding this!

A few general thoughts/questions:
* Would it be better to have one tutorial for the scalar plugin as opposed to multiple (e.g. include gradient tape here as well)? Or maybe it's worth breaking things down a bit, just need to make sure the titles match (gradient tape can also be used with Keras).
* Is there more information that might be valuable to include in this tutorial? 
    * For example, explaining the left pane of scalars? Probably don't need to go into too much detail. 
    * Additionally, explaining that you can add custom Keras metrics and they will appear in the callback? We don't need to explain everything but should give a sense of all the different capabilities here. WDYT?

**Comment by manivaradarajan** [comment]
Thanks for the feedback. I've added an example of custom scalars. I decided that using GradientTape would make it very complicated. Hopefully this complements the original material.

File: docs/r2/tensorboard_scalars_and_keras_basic.ipynb [hunk 1]
```diff
@@ -0,0 +1,452 @@
+{
+  "nbformat": 4,
+  "nbformat_minor": 0,
+  "metadata": {
+    "colab": {
+      "name": "tensorboard_scalars_and_keras_basic.ipynb",
```
Comment (GalOshri) [inline docs/r2/tensorboard_scalars_and_keras_basic.ipynb]: Is this still "basic" now that custom scalars are included? Would there be a more advanced tutorial?

File: docs/r2/tensorboard_scalars_and_keras_basic.ipynb [hunk 2]
```diff
@@ -0,0 +1,452 @@
...
+      "metadata": {
+        "id": "eDXRFe_qp5C3",
+        "colab_type": "text"
+      },
+      "cell_type": "markdown",
+      "source": [
+        "\n",
+        "## Overview\n",
+        "\n",
+        "A basic task in machine learning is understanding how key metrics such as loss change as training progresses. These metrics can help you understand if you're [overfitting](https://en.wikipedia.org/wiki/Overfitting), for example, or if you're unnecessarily training for too long. You may want to compare these metrics across different training runs to help debug and improve your model.\n",
```
Comment (GalOshri) [inline docs/r2/tensorboard_scalars_and_keras_basic.ipynb]: I often think of "task" in ML as something like "classification" or "regression" (the problem being solved). Perhaps we call it a requirement?

File: docs/r2/tensorboard_scalars_and_keras_basic.ipynb [hunk 3]
```diff
@@ -0,0 +1,452 @@
...
+        "# Create some input data between -1 and 1 and randomize it.\n",
+        "X = np.linspace(-1, 1, data_size)\n",
+        "np.random.shuffle(X)\n",
+        "\n",
+        "# Generate the output data.\n",
+        "# f(X) = 0.5X + 2 + noise\n",
+        "Y = 0.5 * X + 2 + np.random.normal(0, 0.05, (data_size, ))\n",
+        "\n",
+        "# Split into test and train pairs.\n",
+        "X_train, Y_train = X[:train_size], Y[:train_size]\n",
```
Comment (GalOshri) [inline docs/r2/tensorboard_scalars_and_keras_basic.ipynb]: Should we standardize on x_train / y_train (instead of X_train / Y_train)? That's what I've seen in the other docs.

File: docs/r2/tensorboard_scalars_and_keras_basic.ipynb [hunk 4]
```diff
@@ -0,0 +1,452 @@
...
+        "colab_type": "text"
+      },
+      "cell_type": "markdown",
+      "source": [
+        "\n",
+        "## Overview\n",
+        "\n",
+        "A basic task in machine learning is understanding how key metrics such as loss change as training progresses. These metrics can help you understand if you're [overfitting](https://en.wikipedia.org/wiki/Overfitting), for example, or if you're unnecessarily training for too long. You may want to compare these metrics across different training runs to help debug and improve your model.\n",
+        "\n",
+        "TensorBoard's **Scalars Plugin** allows you to visualize these metrics using a simple API with very little effort. These visualizations are called **Scalar Summaries**. \n",
```
Comment (GalOshri) [inline docs/r2/tensorboard_scalars_and_keras_basic.ipynb]: I learned from my PR that for user-facing docs we should use "dashboard" instead of "plugin".

Comment (GalOshri) [inline docs/r2/tensorboard_scalars_and_keras_basic.ipynb]: Are the visualizations called scalar summaries? It's a bit confusing as the API is called summary. Was this meant to say the metrics are logged using scalar summary APIs?

File: docs/r2/tensorboard_scalars_and_keras_basic.ipynb [hunk 5]
```diff
@@ -0,0 +1,452 @@
...
+    },
+    {
+      "metadata": {
+        "id": "6pck56gKReON",
+        "colab_type": "code",
+        "colab": {}
+      },
+      "cell_type": "code",
+      "source": [
+        "%tensorboard --logdir logs/scalars"
```
Comment (GalOshri) [inline docs/r2/tensorboard_scalars_and_keras_basic.ipynb]: In a text cell under TensorBoard, add an html comment that points to an image of what the static docs should show. Example [here](https://github.com/tensorflow/tensorboard/pull/1870/files#diff-05296e0e7eafc631c60495988e592d11R240).

File: docs/r2/tensorboard_scalars_and_keras_basic.ipynb [hunk 6]
```diff
@@ -0,0 +1,452 @@
...
+    {
+      "metadata": {
+        "id": "6YDAoNCN3ZNS",
+        "colab_type": "text"
+      },
+      "cell_type": "markdown",
+      "source": [
+        "## Set up data for a simple regression\n",
+        "\n",
+        "You're now going to use Keras to calculate a regression, i.e., find the best line of fit for a paired data set. (While using neural networks and gradient descent is [overkill for this kind of problem](https://https://stats.stackexchange.com/questions/160179/do-we-need-gradient-descent-to-find-the-coefficients-of-a-linear-regression-mode), it does make for a very easy to understand example.)\n",
```
Comment (GalOshri) [inline docs/r2/tensorboard_scalars_and_keras_basic.ipynb]: Typo in URL.

File: docs/r2/tensorboard_scalars_and_keras_basic.ipynb [hunk 7]
```diff
@@ -0,0 +1,452 @@
...
+      },
+      "cell_type": "markdown",
+      "source": [
+        "You may see TensorBoard say that \"No dashboards are active for the current data set\". That's because initial logging data hasn't been saved yet. As training progresses, the Keras model will start logging data. TensorBoard will periodicially refresh and show you your scalar metrics. If you're impatient, you can tap the Refresh arrow on the top right.\n",
+        "\n",
+        "As you watch the training progress, note how both training and validation loss rapidly decrease and then remain stable. In fact, you could have stopped training after 25 epochs, because the training didn't improve much after that point.\n",
+        "\n",
+        "Hover over the graph to see specific data points. You can also try zooming in with your mouse, or selecting part of them to view more detail.\n",
+        "\n",
+        "Using the \"Runs\" selector on the left, you can select specific runs, or select only training or validation. As you continue to experiment and develop your model, you can compare runs by selecting only those you want to examine.\n"
```
Comment (GalOshri) [inline docs/r2/tensorboard_scalars_and_keras_basic.ipynb]: Briefly explain what is a "run" (probably in one of the previous paragraphs)

File: docs/r2/tensorboard_scalars_and_keras_basic.ipynb [hunk 8]
```diff
@@ -0,0 +1,452 @@
...
+      ]
+    },
+    {
+      "metadata": {
+        "id": "finK0GfYyefe",
+        "colab_type": "text"
+      },
+      "cell_type": "markdown",
+      "source": [
+        "Ok, TensorBoard indicates that the metrics look good! Now see how the model actually behaves in real life. \n",
```
Comment (GalOshri) [inline docs/r2/tensorboard_scalars_and_keras_basic.ipynb]: Is there anything we can say about why the metrics look good? How did we judge that?

File: docs/r2/tensorboard_scalars_and_keras_basic.ipynb [hunk 9]
```diff
@@ -0,0 +1,452 @@
...
+        "What if you want to log custom values, such as a dynamic learning rate? To do that, you need to use the TensorFlow Summary API.\n",
+        "\n",
+        "Retrain the regression model and log a custom learning rate. Here's how:\n",
+        "\n",
+        "1.  Create a file writer, using ```tf.summary.create_file_writer()```.\n",
+        "2.  Define a custom learning rate function. This will be passed to the Keras [LearningRateScheduler](https://www.tensorflow.org/api_docs/python/tf/keras/callbacks/LearningRateScheduler) callback.\n",
+        "3.  Inside the learning rate function, use ```tf.summary.scalar()``` to log the custom learning rate.\n",
+        "4.  Pass the LearningRateScheduler callback to Model.fit().\n",
+        "\n",
+        "In general, to log a custom scalar, you need to use ```tf.summary.scalar()``` with a file writer. You can reuse the file writer for multiple scalars."
```
Comment (GalOshri) [inline docs/r2/tensorboard_scalars_and_keras_basic.ipynb]: The file writer corresponds to a specific run, right?

File: docs/r2/tensorboard_scalars_and_keras_basic.ipynb [hunk 10]
```diff
@@ -0,0 +1,452 @@
...
+      "outputs": []
+    },
+    {
+      "metadata": {
+        "id": "RRlUDnhlkN_q",
+        "colab_type": "text"
+      },
+      "cell_type": "markdown",
+      "source": [
+        "Using the \"Runs\" selector on the left, notice that you have a ```<timestamp>/metrics``` run. Selecting this show a \"learning rate\" graph that allows you to verify the progression of the learning rate during this run. \n",
```
Comment (GalOshri) [inline docs/r2/tensorboard_scalars_and_keras_basic.ipynb]: Typo: "selecting this show a "

File: docs/r2/tensorboard_scalars_and_keras_basic.ipynb [hunk 11]
```diff
@@ -0,0 +1,452 @@
...
+    {
+      "metadata": {
+        "id": "vvwGmJK9XWmh",
+        "colab_type": "text"
+      },
+      "cell_type": "markdown",
+      "source": [
+        "## Logging custom scalars\n",
+        "\n",
+        "What if you want to log custom values, such as a dynamic learning rate? To do that, you need to use the TensorFlow Summary API.\n",
```
Comment (GalOshri) [inline docs/r2/tensorboard_scalars_and_keras_basic.ipynb]: I like this example!

File: docs/r2/tensorboard_scalars_and_keras_basic.ipynb (hunk 1 above)
Comment (manivaradarajan) [reply docs/r2/tensorboard_scalars_and_keras_basic.ipynb]: Renamed.

File: docs/r2/tensorboard_scalars_and_keras_basic.ipynb (hunk 2 above)
Comment (manivaradarajan) [reply docs/r2/tensorboard_scalars_and_keras_basic.ipynb]: Reworded

File: docs/r2/tensorboard_scalars_and_keras_basic.ipynb (hunk 3 above)
Comment (manivaradarajan) [reply docs/r2/tensorboard_scalars_and_keras_basic.ipynb]: Fixed.

File: docs/r2/tensorboard_scalars_and_keras_basic.ipynb (hunk 4 above)
Comment (manivaradarajan) [reply docs/r2/tensorboard_scalars_and_keras_basic.ipynb]: Renamed

Comment (manivaradarajan) [reply docs/r2/tensorboard_scalars_and_keras_basic.ipynb]: Reworded.

File: docs/r2/tensorboard_scalars_and_keras_basic.ipynb (hunk 10 above)
Comment (manivaradarajan) [reply docs/r2/tensorboard_scalars_and_keras_basic.ipynb]: Fixed

File: docs/r2/tensorboard_scalars_and_keras_basic.ipynb (hunk 11 above)
Comment (manivaradarajan) [reply docs/r2/tensorboard_scalars_and_keras_basic.ipynb]: :-)

File: docs/r2/tensorboard_scalars_and_keras_basic.ipynb (hunk 6 above)
Comment (manivaradarajan) [reply docs/r2/tensorboard_scalars_and_keras_basic.ipynb]: Fixed

File: docs/r2/tensorboard_scalars_and_keras_basic.ipynb (hunk 9 above)
Comment (manivaradarajan) [reply docs/r2/tensorboard_scalars_and_keras_basic.ipynb]: It doesn't correspond to a run; it's a directory. Reworded.

File: docs/r2/tensorboard_scalars_and_keras_basic.ipynb (hunk 8 above)
Comment (manivaradarajan) [reply docs/r2/tensorboard_scalars_and_keras_basic.ipynb]: Reworded.

File: docs/r2/tensorboard_scalars_and_keras_basic.ipynb (hunk 7 above)
Comment (manivaradarajan) [reply docs/r2/tensorboard_scalars_and_keras_basic.ipynb]: Done.

File: docs/r2/tensorboard_scalars_and_keras_basic.ipynb (hunk 5 above)
Comment (manivaradarajan) [reply docs/r2/tensorboard_scalars_and_keras_basic.ipynb]: Done.

**Comment by manivaradarajan** [comment]
Addressed all comments, PTAL

**Comment by manivaradarajan** [comment]
Addressed all comments, PTAL
